extras/data/machinery/generate/utils.py

```diff
@@ -81,15 +81,34 @@ def dump_polarized(src_path, target):
         observables dictionary, corresponding to the runcard
 
     """
+
+    # We make a case distinction for several experiments:
     obs = obs_template.copy()
-    data = load(src_path, 0, ["x", "Q2"])
-    dict_kins = [
-        dict(x=d["x"]["mid"], y=d["y"]["mid"], Q2=d["Q2"]["mid"]) for d in data
-    ]
+    data = np.genfromtxt(src_path)
+
+    if (
+        "ATHENA" in target.parent.name
+        or "EIC" in target.parent.name
+        or "EIcC" in target.parent.name
+    ):
+        dict_kins = [dict(x=float(d[0]), y=0.0, Q2=float(d[1])) for d in data]
+
+    else:
+        data = load(src_path, 0, ["x", "Q2"])
+        dict_kins = [
+            dict(x=d["x"]["mid"], y=d["y"]["mid"], Q2=d["Q2"]["mid"]) for d in data
+        ]
+
+    heavyness = "total"
+    if "EIC" in target.parent.name or "EIcC" in target.parent.name:
+        heavyness = "charm"
```

Comment:
again, better but not good enough yet :upside_down_face: we can save these hard coded lines ... 1) we should always aim to avoid hardcoded stuff, as this will hinder the extensibility greatly 2) having this `if` here make the EIC implementation not self-contained, i.e. part of the logic is sitting here;

let me repeat what I said: "let's make `heavyness` an argument`; this means
- change the signature of this function from `dump_polarized(src_path, target)` to `dump_polarized(src_path, target, heavyness="total")`
- and then in the driver file you replace `from blub.bla import dump` by `from blub.bla import dump as raw_dump [...] def dump(src_path, arget): return raw_dump(src, target, "charm")`